src/assembler/instruction.h

```diff
@@ -103,14 +103,15 @@ struct Opcode : public Field {
 };
 
 struct Reg : public Field {
+    enum class Regsd { rd, rs1, rs2 };
     /**
      * @brief Reg
      * @param tokenIndex: Index within a list of decoded instruction tokens that corresponds to the register index
      * @param range: range in instruction field containing register index value
      */
-    Reg(const ISAInfoBase* isa, unsigned tokenIndex, BitRange range) : Field(tokenIndex), m_range(range), m_isa(isa) {}
-    Reg(const ISAInfoBase* isa, unsigned tokenIndex, unsigned _start, unsigned _stop)
-        : Field(tokenIndex), m_range({_start, _stop}), m_isa(isa) {}
+    Reg(const ISAInfoBase* isa, unsigned tokenIndex, BitRange range, Regsd _regsd) : Field(tokenIndex), m_range(range), m_isa(isa), regsd(_regsd) {}
```

Comment:
As above, regsd should be a string. Also, there could be a default argument here, for instance "r" or "reg"